src/classes/RD2_ScheduleService_TEST.cls

```diff
@@ -1373,8 +1373,105 @@ private with sharing class RD2_ScheduleService_TEST {
 
     }
 
+    /****
+    * @description Validates that 1 active schedule is returned for a new RD
+    */
+    @IsTest
+    private static void shouldReturn1ActiveSchedulesForNewRD() {
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact contact = getContact();
+
+        npe03__Recurring_Donation__c rd =
+                TEST_RecurringDonationBuilder.constructEnhancedBuilder()
```

Comment:
Indentation is off. It might be better to extract common RD builder setting in a helper method and then call it throughout the test class and add/modify changes where required. 

Also, is is better to be explicit in test name and use `One` instead of `1` (it looks like `l` in text). Let's validate the `ActiveSchedule` vars.

```
    /****
    * @description Validates that one active schedule is returned for a new RD
    */
    @IsTest
    private static void shouldReturnActiveScheduleWhenRDIsCreated() {
        RD2_EnablementService_TEST.setRecurringDonations2Enabled();

        npe03__Recurring_Donation__c rd = getRecurringDonationMonthlyBuilder()
            .build();

        Test.startTest();
        insert rd;
        Test.stopTest();
        
        RD2_ScheduleService service = new RD2_ScheduleService();
        List<RD2_ScheduleService.ActiveSchedule> schedules = service.getVisualizedSchedules(rd);
        System.assertEquals(1, schedules.size(), 'There should be one schedule: ' + schedules);
    }
 
    private static TEST_RecurringDonationBuilder getRecurringDonationWeeklyBuilder() {
        return TEST_RecurringDonationBuilder.constructEnhancedBuilder()
            .withMockId()
            .withContact(UTIL_UnitTestData_TEST.mockId(Contact.SObjectType))
            .withAmount(400)
            .withPaymentMethod('Check')
            .withInstallmentPeriodWeekly()
            .withInstallmentFrequency(1)
            .withDateEstablished(Date.newInstance(2019, 9, 15))
            .withStartDate(Date.newInstance(2018, 3, 17));
    }

    private static TEST_RecurringDonationBuilder getRecurringDonationMonthlyBuilder() {
        return TEST_RecurringDonationBuilder.constructEnhancedBuilder()
            .withContact(getContact().Id)
            .withAmount(50)
            .withPaymentMethod('Check')
            .withInstallmentPeriodMonthly()
            .withDayOfMonth('12')
            .withInstallmentFrequency(1)
            .withDateEstablished(Date.today())
            .withStartDate(Date.today())
            .build();
    }
```

Reply:
Let's update tests and verify return values (note there is a comment to move `active schedule in the visualizer controller not in the schedule service).